ConfigSpace/hyperparameters.pyx

```diff
@@ -151,6 +151,36 @@ cdef class Hyperparameter(object):
     cpdef int compare_vector(self, DTYPE_t value, DTYPE_t value2):
         raise NotImplementedError()
 
+    def pdf(self, vector: np.ndarray) -> np.ndarray:
+        """
+        Computes the probability density function of the parameter in 
+        the original parameter space (the one specified by the user).
+        For each parameter type, there is also a method _pdf which 
+        operates on the transformed (and possibly normalized) parameter
+        space. Only legal values return a positive probability density,
+        otherwise zero.
+        
+        Parameters
+        ----------
+        vector: np.ndarray
+            the (N, ) vector of imputs for which the probability density
+            function is to be computed.
+        """
+        raise NotImplementedError()
```

Comment:
@mfeurer I had to revisit this, since I thought about this quite a lot myself when deciding on the design.

Your suggestion is pretty much what I wanted to do. However, the _inverse_transform in constants and ordinals do not make sense with it, in my opinion. Therefore, I got a bit hesitant about the generality of the solution you're proposing. 
However, the workaround in this case is easy (or we just overwrite in that specific case, since the probability for ordinals is presumed uniform anyway, as of now). If you want to change it, I'm happy to do so!

The default would be to do like in FloatHP:
`        if vector.ndim != 1:`
            `raise ValueError("Method pdf expects a one-dimensional numpy array")`
        `vector = self._inverse_transform(vector)`
        `return self._pdf(vector)`